community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -545,9 +520,10 @@ public long nextId()
      */
     public void freeId( long id )
     {
-        if ( idGenerator != null )
+        IdGenerator generator = this.idGenerator;
```

Comment:
So the value we null-check and the object we call a method on are always the same.